File: frontend/lib/services/interfaces/IAuthService.ts
/**
 * Auth Service Interface
 * Defines business logic contract for authentication
 */

import { Result } from '../../types/common.types'
import { AuthResponse, GoogleAuthRequest } from '../../types/auth.types'

export interface IAuthService {
  authenticateWithGoogle(request: GoogleAuthRequest): Promise<Result<AuthResponse>>
  checkServiceHealth(): Promise<Result<{ status: string; service: string }>>
}

File: backend/Services/Interfaces/IAuthService.cs
using backend.Common;
using backend.DTOs.Auth.Request;
using backend.DTOs.Auth.Response;

namespace backend.Services.Interfaces;

/// <summary>
/// Authentication service interface
/// </summary>
public interface IAuthService
{
    Task<Result<AuthResponse>> AuthenticateWithGoogleAsync(GoogleAuthRequest request, CancellationToken cancellationToken = default);
}

